File: Assets/Scripts/Javascript/LightSwitch.js
#pragma strict

public var targetLight:GameObject;
private var playerInReach:boolean;

function Update() {
	if(playerInReach && Input.GetKeyDown("e")) {
		targetLight.SendMessage("ToggleLight");
	}
}

function OnTriggerEnter(collider:Collider) {
	if(collider.gameObject.tag == "Player")
		playerInReach = true;
	
}

function OnTriggerExit(collider:Collider) {
	if(collider.gameObject.tag == "Player")
		playerInReach = false;
}

File: Assets/Scripts/C#/LightSwitch.cs

using UnityEngine;
using System.Collections;

public class LightSwitch : MonoBehaviour {
	
	
	public GameObject targetLight;
	private bool  playerInReach;
	
	void  Update (){
		if(playerInReach && Input.GetKeyDown("e")) {
			targetLight.SendMessage("ToggleLight");
		}
	}
	
	void  OnTriggerEnter ( Collider collider  ){
		if(collider.gameObject.tag == "Player")
			playerInReach = true;
		
	}
	
	void  OnTriggerExit ( Collider collider  ){
		if(collider.gameObject.tag == "Player")
			playerInReach = false;
	}
}